src/functions-and-arrays.js

```diff
@@ -1,81 +1,118 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(number1, number2) {
+    if (typeof number1 != 'number' || typeof number2 != 'number') {
+        return;
+    }
+    
+    if (number1 > number2) {
+        return number1;
+    } else {
+        return number2;
+    }
+}
+maxOfTwoNumbers(4, 9);
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(wordNames) {
+    if (wordNames.length === 0) {
+        return null;
+    }
+
+    wordNames.splice(1, 2, 3, 4, 5, 6);
+    return wordNames;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(plus) {      
+    let total = 0;
+    plus.forEach(number => total += number);
+    return total;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(plus) {
+    let total = 0;
+    plus.forEach(number => total += number);
+    return total;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(calculate) {
+    let result = 0;
+    calculate.forEach(number => result += number);
+    return result;
+}
```

Comment:
- Aquí, al parámetro de la función lo has llamado `calculate`, pero te recomiendo que pongas nombres descriptivos sobre lo que se le está pasando a la función, en este caso te serviría `array` o `numbers`.
- También te falta cubrir el caso de si el array está vacío, lo cual se suplía con la línea `if(calculate.length === 0 { return null })`
- Además, estás devolviendo directamente el `result` (que es la suma), pero te están pidiendo la media, por lo que ese total lo deberías dividir entre el número de elementos del array (que se puede calcular a través de su `length`)

```suggestion
const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];

function averageNumbers(array) {

    if (array.length === 0) {
        return null
    }

    let result = 0;
    array.forEach(number => result += number);
    return result / array.length;
}
```